app/src/main/java/com/example/farmeraid/data/FarmRepository.kt

```diff
@@ -0,0 +1,68 @@
+package com.example.farmeraid.data
+
+import android.util.Log
+import com.example.farmeraid.data.model.FarmModel
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.FieldValue
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.tasks.await
+
+class FarmRepository(
+    private val userRepository: UserRepository
+) {
+    private var db: FirebaseFirestore = FirebaseFirestore.getInstance()
+
+    suspend fun createFarm(farmName: String){
+        //if db.collection("farm").get
+        try {
+            userRepository.getUserId()?.let{
+                val docRef = db.collection("farm").add(
+                    mapOf (
+                        "name" to farmName,
+                        "users" to listOf(it),
+                        "markets" to emptyList<String>(),
+                        "charities" to emptyList<String>()
+                    )
+                ).await()
+
+                db.collection("inventory").document(docRef.id).set(
+                    mapOf("produce" to emptyMap<String,Int>())
+                )
+
+                //TODO create new transaction document for the farm
+
+                //TODO update userRespository with new farmID
+                //TODO Function needs to be made on UserRepo
+
+            }
+        }catch(e: Exception){
+            Log.e("FarmRepository - createFarm()", e.message?:"Unknown error")
+        }
+
+
+
+
+    }
+    suspend fun joinFarm(farmCode: String){
+        try {
+            userRepository.getUserId()?.let{id ->
+                db.collection("farm").let{ ref ->
+                    ref.whereEqualTo("code", farmCode).get().await().let{ snap ->
```

Comment:
I think what we could do instead of a code property is use the farmID as the code. IE) if the user provides the farmID they join that particular farm if it exits. 

This way it makes it much simpler from a code perspective. We do not want to add extra logic of expiry codes and stuff right now. What do you think?

Reply:
we were thinking about this, but a few things to consider:
- farmID could be complex, so not the best for user to input
- if the code isn't complex could potentially not be good for scaling
- Having the farmID as the code could also be a security issue since we are exposing the ID that is used by our database

Reply:
I see, how would we generate this code?

Reply:
I think we were thinking of doing some sort of UUID or some other generator (like a random 6 character generator) it would have to be done on creation of a Farm. But that would probably be its own PR

Reply:
Yeah makes sense, so is the idea is to generate a random 6 digit code per farm and they use that to join. If so I like that idea, we def can use a lib or sum for that